File: 2507-number-of-common-factors/2507-number-of-common-factors.cpp
class Solution {
public:
    int commonFactors(int a, int b) {
        int factors = 1, n = min(a,b);
        for(int x = 2; x <= n; x++)
            if(a % x == 0 && b % x == 0)
                factors++;
        return factors;
    }
};

File: 2507-number-of-common-factors/2507-number-of-common-factors.py
class Solution:
    def commonFactors(self, a: int, b: int) -> int:
        factors = 1
        n = math.gcd(a,b)
        for i in range(2,n+1):
            if a % i == 0 and b % i == 0:
                factors += 1
        return factors